src/libraries/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs

```diff
@@ -560,6 +560,77 @@ public async Task WriteAsync_MultipleConcurrentValueTaskWrites_AsTask_Success()
             }
         }
 
+        [Fact]
+        public async Task NetworkStream_ReadTimeout_RemainUseable()
+        {
+            using StreamPair streams = await CreateConnectedStreamsAsync();
+            NetworkStream readable = (NetworkStream)streams.Stream1;
+
+            Assert.True(readable.Socket.Connected);
+            readable.Socket.ReceiveTimeout = TestSettings.FailingTestTimeout;
+            var buffer = new byte[100];
+            int readBytes;
+            try
+            {
+                readBytes = readable.Read(buffer);
+            }
+            catch (IOException ex) when (ex.InnerException is SocketException && ((SocketException)ex.InnerException).SocketErrorCode == SocketError.TimedOut)
+            {
+            }
+            Assert.True(readable.Socket.Connected);
+
+            try
+            {
+                readBytes = readable.Read(buffer);
+            }
+            catch (IOException ex) when (ex.InnerException is SocketException && ((SocketException)ex.InnerException).SocketErrorCode == SocketError.TimedOut)
+            {
+            }
+            Assert.True(readable.Socket.Connected);
+
+            streams.Stream2.Write(new byte[] { 65 });
+            readBytes = readable.Read(buffer);
+            Assert.Equal(1, readBytes);
+            Assert.True(readable.Socket.Connected);
+        }
```

Comment:
If I've understood correctly, the above test passes before this PR and the below test only passes after this PR?

Reply:
yes. This is because of the existing `errorCode != SocketError.TimedOut`.